lint.go

```diff
@@ -713,3 +714,83 @@ func CheckEarlyDefer(f *lint.File) {
 	}
 	f.Walk(fn)
 }
+
+func CheckEmptyCriticalSection(f *lint.File) {
+	fn := func(node ast.Node) bool {
+		block, ok := node.(*ast.BlockStmt)
+		if !ok {
+			return true
+		}
+		if len(block.List) < 2 {
+			return true
+		}
+		for i := range block.List {
+			if i == len(block.List)-1 {
+				break
+			}
+
+			mutexParams := func(s ast.Stmt) ([]string, string, bool) {
```

Comment:
Define this function outside of the loop. If it needs to access `i`, pass that in as an argument.